### PR TITLE
First draft strategy

### LINKED ISSUE #4: write down a proposition and options for our strategy to approach the estimation

### PR DESCRIPTION
**Pull request by pdobacz**
closes #4, these are just first draft thoughts on how we could structure our investigations.

### CODE REVIEW

File: docs/strategy.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
+# Gas Cost Estimator approach strategy
+
+Here we organize thoughts on different approaches to consider.
+We focus on EVM for simplicity, but this should be substitutable by eWASM throughout the document.
```
Comment (rodiazet) [inline docs/strategy.md:4]: Ethereum call it just WASM

File: docs/strategy.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+Q: how does the measurement method impact the requirements for program generation? Can we decouple those?
+
+Q: will manual, "best guess" generation of sample programs do, or should it be a result of some feedback loop (see also _Instrumentation and measurement_)
+    - probably, if we find the "measure one" approach to give good results, manual will be good. Similar with "measure all", since it still measures instructions individually.
+    - if we find that instructions cannot be reliably measured individually, and we have to revert to "measure total", it is likely that some algorithmic way of generating the sample program set is necessary.
+
+### Additional ideas
+
+1. "Broken metre" paper does a genetic algorithm minimizing the gas throughput (gas/second).
```
Comment (rodiazet) [inline docs/strategy.md:45]: Can we add link to this paper?

**Review by rodiazet — CHANGES_REQUESTED** [review]

File: docs/strategy.md (hunk 1 above)
Comment (pdobacz) [reply docs/strategy.md:4]: i've seen ewasm is quite common, let's be specific. https://github.com/ewasm but I'll correct the spelling :)

**Comment by pdobacz** [comment]
approved during call, merging

**Pull request merged**